pkg/agent/controller/traceflow/packetin.go

```diff
@@ -70,13 +71,16 @@ func (c *Controller) parsePacketIn(pktIn *ofctrl.PacketIn) (*opsv1alpha1.Tracefl
 	var match *ofctrl.MatchField
 
 	// Get data plane tag.
-	if match = getMatchRegField(matchers, uint32(openflow.TraceflowReg)); match == nil {
-		return nil, nil, errors.New("traceflow data plane tag not found")
-	}
-	rngTag := openflow13.NewNXRange(int(openflow.OfTraceflowMarkRange[0]), int(openflow.OfTraceflowMarkRange[1]))
-	tag, err := getInfoInReg(match, rngTag)
-	if err != nil {
-		return nil, nil, err
+	// Directly read data plane tag from packet.
+	var tag uint8
+	if pktIn.Data.Ethertype == protocol.IPv4_MSG {
+		ipPacket, ok := pktIn.Data.Data.(*protocol.IPv4)
+		if !ok {
+			return nil, nil, errors.New("invalid traceflow IPv4 packet")
+		}
+		tag = ipPacket.DSCP
+	} else {
+		return nil, nil, errors.New(fmt.Sprintf("unsupported traceflow packet Ethertype: %d", pktIn.Data.Ethertype))
```

Comment:
we typically use `fmt.Errorf("unsupported traceflow packet Ethertype: %d", pktIn.Data.Ethertype)` for this instead of `errors.New` + `fmt.Sprintf`

Reply:
Fixed. Thanks.